api/v1alpha1/amaltheasession_types.go

```diff
@@ -0,0 +1,283 @@
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+// Important: Run "make" to regenerate code after modifying this file
+
+// AmaltheaSessionSpec defines the desired state of AmaltheaSession
+type AmaltheaSessionSpec struct {
+	// Specification for the main session container that the user will access and use
+	Session Session `json:"session"`
+
+	// A list of code repositories and associated configuration that will be cloned in the session
+	CodeRepositories []CodeRepository `json:"codeRepositories,omitempty"`
+
+	// A list of data sources that should be added to the session
+	DataSources []DataSource `json:"dataSources,omitempty"`
+
+	// Authentication configuration for the session
+	Authentication Authentication `json:"authentication,omitempty"`
+
+	// Culling configuration
+	Culling Culling `json:"culling,omitempty"`
+
+	// Will hibernate the session, scaling the session's statefulset to zero.
+	Hibernated bool `json:"hibernated,omitempty"`
+
+	// +kubebuilder:default:=false
+	// Whether to adopt all secrets referred to by name in this CR. Adopted secrets will be deleted when the CR is deleted.
+	AdoptSecrets bool `json:"adoptSecrets,omitempty"`
+
+	// Additional containers to add to the session statefulset.
+	// NOTE: The container names provided will be partially overwritten and randomized to avoid collisions
+	ExtraContainers []v1.Container `json:"extraContainers,omitempty"`
+
+	// Additional init containers to add to the session statefulset
+	// NOTE: The container names provided will be partially overwritten and randomized to avoid collisions
+	ExtraInitContainers []v1.Container `json:"initContainers,omitempty"`
+
+	// Configuration for an ingress to the session, if omitted a Kubernetes Ingress will not be created
+	Ingress Ingress `json:"ingress,omitempty"`
+}
+
+type Session struct {
+	Image string `json:"image"`
+	// The command to run in the session container, if omitted it will use the Docker image ENTRYPOINT
+	Command []string `json:"command,omitempty"`
+	// The arguments to run in the session container, if omitted it will use the Docker image CMD
+	Args []string    `json:"args,omitempty"`
+	Env  []v1.EnvVar `json:"env,omitempty"`
+	// Resource requirements and limits in the same format as a Pod in Kubernetes
+	Resources v1.ResourceRequirements `json:"resources,omitempty"`
+	// +kubebuilder:default:=8000
+	// +kubebuilder:validation:ExclusiveMinimum:=true
+	// +kubebuilder:validation:Minimum:=0
+	// The TCP port where whatever is running in the session container will listen on for connections
+	Port    int32   `json:"port"`
+	Storage Storage `json:"storage,omitempty"`
+	// The abolute path for the working directory of the session container, if omitted it will use the image
+	// working directory.
+	WorkingDir string `json:"workingDir,omitempty"`
+	// +kubebuilder:default:=1000
+	// +kubebuilder:validation:Minimum:=0
+	RunAsUser int `json:"runAsUser"`
```

Comment:
To match original type
```suggestion
	RunAsUser int64 `json:"runAsUser"`
```